qa/workunits/ceph-helpers.sh

```diff
@@ -1048,6 +1048,44 @@ function test_repair() {
     ! TIMEOUT=1 repair 1.0 || return 1
     teardown $dir || return 1
 }
+#######################################################################
+
+##
+# Run scrub on **pgid** and wait until it completes. The repair
+# function will fail if repair does not complete within $TIMEOUT
+# seconds. The scrub is complete whenever the
+# **get_last_scrub_stamp** function reports a timestamp different from
+# the one stored before starting the scrub.
+#
+# @param pgid the id of the PG
+# @return 0 on success, 1 on error
+#
+function pg_scrub() {
+    local pgid=$1
+    local last_scrub=$(get_last_scrub_stamp $pgid)
+
+    ceph pg scrub $pgid
+    for ((i=0; i < $TIMEOUT; i++)); do
+        if test "$last_scrub" != "$(get_last_scrub_stamp $pgid)" ; then
+            echo "pg $pgid scrub finish."
+            return 0
+        fi
+        sleep 1
```

Comment:
there still is a window of opportunity for ceph -w to be killed before it had a chance to issue the last meaningfull information about scrub. I have to agree this is a border case that is unlikely to happen. But this kind of problem are also the one that drive me nuts when I run into them months later.

what about greping the log for lines that contain "scrub" so the noise is reduced ? I'm not too concerned about noise since the output only shows when an error happens. So it will not polute the output when an unrelated test fail.